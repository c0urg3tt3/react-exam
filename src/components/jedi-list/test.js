import React from 'react'
import { shallow } from 'enzyme'

import JediList from './component'

describe('[component] JediList', () => {
  it('should render component', () => {
    const wrapper = shallow(<JediList/>)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('ul')).toEqual(true)
    expect(wrapper.hasClass('jedi-list')).toEqual(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render children as function', () => {
    const jedies = [
      { id: 1337, name: 'Anakin Skywalker' },
      { id: 4242, name: 'Sheev Palpatine' }
    ]

    const wrapper = shallow(
      <JediList isFetching={true}>
      {
        ({jedies}) => jedies.map((jedi, index) => {
          return (<li key={index} className="jedi">{jedi.name}</li>)
        })
      }
      </JediList>
    )

    expect(wrapper.find('li')).toHaveLength(0)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ jedies })

    expect(wrapper.find('li')).toHaveLength(2)
    expect(wrapper.find('li').first().text()).toEqual(jedies[0].name)
    expect(wrapper.find('li').last().text()).toEqual(jedies[1].name)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
