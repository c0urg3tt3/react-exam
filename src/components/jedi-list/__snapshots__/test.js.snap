// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] JediList should render children as function 1`] = `"<ul class=\\"jedi-list\\"></ul>"`;

exports[`[component] JediList should render children as function 2`] = `"<ul class=\\"jedi-list\\"><li class=\\"jedi\\">Anakin Skywalker</li><li class=\\"jedi\\">Sheev Palpatine</li></ul>"`;

exports[`[component] JediList should render component 1`] = `"<ul class=\\"jedi-list\\"></ul>"`;
